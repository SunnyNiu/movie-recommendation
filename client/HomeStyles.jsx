import styled from 'styled-components'
import { Link } from 'react-router-dom'

export const BackToHomeButton = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0 1em;
  font-size: 2rem;
  text-align: center;
  &:hover {
    cursor: pointer;
    color: black;
  }
`

export const StartButton = styled.button`
    border: 2px solid palevioletred;
    color: palevioletred;
    margin: 0 1em;
    font-size: 2rem;
  &:hover {
    cursor: pointer;
    color: black;
  }
`

export const LinkContainer = styled(Link)`
height: 100%;
align-items: center;
font-size: 1.5rem;
font-style: italic;
`
export const WelcomeTitle = styled.p`
font-style: italic;
font-size: 2rem;
font-family: Arial, Helvetica, sans-serif;
align-items: center;
margin: 2em 0em
`
