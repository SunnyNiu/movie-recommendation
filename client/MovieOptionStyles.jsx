import styled from 'styled-components'

export const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0em 1em;
  font-size: 1rem;
  text-align: center;
  &:hover {
    cursor: pointer;
    color: black;
  }
`