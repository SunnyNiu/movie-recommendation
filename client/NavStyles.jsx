import styled from 'styled-components'

export const HeaderContainer = styled.div`
height: 70px;
width: 100%;
display: flex;
justify-content: space-between;
background-color:  #323c50 !important;
font-size:1.5rem;
color: #fff !important; 
font-family: Arial, Helvetica, sans-serif;
line-height: 3
`
