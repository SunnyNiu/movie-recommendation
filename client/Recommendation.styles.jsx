import styled from 'styled-components'

export const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0em 1em;
  font-size: 1rem;
  text-align: center;
  &:hover {
    cursor: pointer;
    color: black;
  }
`
export const MovieContainer = styled.div`
display: flex;
flex-direction: column;
align-items: center;
`

export const Container = styled.div`
`

export const ButtonContainer = styled.div`
`
export const Img = styled.img`
  background-size: cover;
  background-position: center;
  margin-bottom: 5px;
  opacity: 0.8;
  max-width:100%;
  max-height: 500px;
`

export const Title = styled.h1`
font-style: italic;
font-size: 1rem;
font-family: Arial, Helvetica, sans-serif;
text-align: center;
align-items: center;
margin: 1em 0em
`
