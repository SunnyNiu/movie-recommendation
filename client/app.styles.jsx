import styled, { keyframes } from 'styled-components'
import { Link } from 'react-router-dom'

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`

export const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0em 1em;
  font-size: 2rem;
  width:250px;
  text-align: center;
  &:hover {
    cursor: pointer;
    color: black;
  }
`

export const BackToHomeButton = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0 1em;
  padding: 0 em 2.5em;
  font-size: 2rem;
  width:350px;
  text-align: center;
  &:hover {
    cursor: pointer;
    color: black;
  }
`

export const StartButton = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 2em 1em;
  padding: 0 em 2.5em;
  font-size: 2rem;
  width:350px;
  text-align: center;
  &:hover {
    cursor: pointer;
    color: black;
  }
`

export const Container = styled.div`
display: flex;
flex-direction: column;
align-items: center;
padding: 500px;
`

export const MovieImg = styled.img`
  height: 500px;
  align-items: center;
`

export const Title = styled.h1`
font-size: 2rem;
font-family: Arial, Helvetica, sans-serif;
height: 120px;
text-align: center;
`

export const HomePageContainer = styled.div`
display: flex;
flex-direction: column;
text-align: center;
padding: 300px;
`

export const MovieContainer = styled.div`
align-items: center;
text-align: center;
`

export const WelcomeTitle = styled.p`
font-style: italic;
font-size: 2.5rem;
font-family: Arial, Helvetica, sans-serif;
align-items: center;
`

export const LinkContainer = styled(Link)`
height: 100%;
width: 70px;
padding: 40px;
align-items: center;
font-size: 1.5rem;
font-style: italic;
`

export const HomeContainer = styled.div`
align-items: center;
background-color: white;
height: 500px;
`
