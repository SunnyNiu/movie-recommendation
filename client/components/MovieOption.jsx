import React from 'react'
import { connect } from 'react-redux'
import { fetchMovie, fetchMovieGenresByMovieId } from '../../redux/movie.actions'
import Recommendation from './Recommendation'
import { Button, Container, MovieImg, Title, MovieContainer } from '../app.styles'

class MovieOption extends React.Component {
  componentDidMount () {
    const { fetchMovie, moviesId } = this.props
    fetchMovie(moviesId)
  }

  render () {
    const { fetchMovieGenresByMovieId, fetchMovie, moviesId, movie: { id, name, image } } = this.props

    return (<Container>
      {moviesId.length > 10 ? (<Recommendation/>) : (
        <div className="notification">
          <MovieContainer>
            <Title>{name}</Title>
            <MovieImg src={image} alt='movieImage'/>
          </MovieContainer>
          <Button
            onClick={() => { fetchMovieGenresByMovieId(id, 'dislike'); fetchMovie(moviesId) }}>
          üëé Dislike
          </Button>
          <Button
            onClick={() => { fetchMovieGenresByMovieId(id, 'like'); fetchMovie(moviesId) }}>
         üëç Like
          </Button>
        </div>
      )}
    </Container>)
  }
}

function mapStateToProps (state) {
  return {
    movie: state.movie,
    genres: state.genres,
    moviesId: state.moviesId
  }
}
const mapDispatchToProps = dispatch => ({
  fetchMovie: (moviesId) => dispatch(fetchMovie(moviesId)),
  fetchMovieGenresByMovieId: (moveId, option) => dispatch(fetchMovieGenresByMovieId(moveId, option))
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieOption)
