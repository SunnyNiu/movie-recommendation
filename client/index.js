import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import reducer from '../redux/movie.reducer'
import { fetchMovie } from '../redux/movie.actions'
import { BrowserRouter } from 'react-router-dom'

import App from './components/App'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(
  reducer,
  composeEnhancers(applyMiddleware(thunk))
)

function render () {
  const root = document.getElementById('app')
  ReactDOM.render(<Provider store={store}><BrowserRouter><App /></BrowserRouter></Provider>, root)
}

render()
store.subscribe(render)
