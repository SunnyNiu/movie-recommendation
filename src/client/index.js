import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';
import reducer from './redux/movie.reducer';
import { HashRouter } from 'react-router-dom';

import App from '../client/components/App';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));

function render() {
  const root = document.getElementById('app');
  ReactDOM.render(
    <Provider store={store}>
      <HashRouter>
        <App />
      </HashRouter>
    </Provider>,
    root
  );
}

render();
store.subscribe(render);
