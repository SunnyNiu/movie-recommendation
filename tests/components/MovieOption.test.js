import React from 'react'
import { mount } from 'enzyme'
import MovieOption from '../../client/components/MovieOption'
import { Button, Title, MovieContainer, ButtonContainer, Img } from '../../client/MovieOptionStyles'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

jest.mock('react-dom')
const mockStore = configureStore([])

describe('<MovieOption /> component tests', () => {
  let store
  let component
  beforeEach(() => {
    store = mockStore({
      movie: '',
      moviesId: [],
      likedMovies: [],
      movies: []
    })

    store.dispatch = jest.fn()

    component = renderer.create(
      <Provider store={store}>
        <MovieOption />
      </Provider>
    )
  })

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should contains text Skip and Like', () => {
    const wrapper = mount(<Provider store={store}>
      <MovieOption />
    </Provider>)

    expect(wrapper.text()).toMatch('ü§≤ Skipüëç Like')
  })

  it('should contains <MovieContainer />', () => {
    const expected = true
    const wrapper = mount(<Provider store={store}>
      <MovieOption />
    </Provider>)

    const actual = wrapper.containsMatchingElement(MovieContainer)
    expect(actual).toBe(expected)
  })

  it('should contains <Button />', () => {
    const expected = true
    const wrapper = mount(<Provider store={store}>
      <MovieOption />
    </Provider>)

    const actual = wrapper.containsMatchingElement(Button)
    expect(actual).toBe(expected)
  })

  it('should contains <Title />', () => {
    const expected = true
    const wrapper = mount(<Provider store={store}>
      <MovieOption />
    </Provider>)

    const actual = wrapper.containsMatchingElement(Title)
    expect(actual).toBe(expected)
  })

  it('should contains <ButtonContainer />', () => {
    const expected = true
    const wrapper = mount(<Provider store={store}>
      <MovieOption />
    </Provider>)

    const actual = wrapper.containsMatchingElement(ButtonContainer)
    expect(actual).toBe(expected)
  })

  it('should contains <Img />', () => {
    const expected = true
    const wrapper = mount(<Provider store={store}>
      <MovieOption />
    </Provider>)

    const actual = wrapper.containsMatchingElement(Img)
    expect(actual).toBe(expected)
  })
})
