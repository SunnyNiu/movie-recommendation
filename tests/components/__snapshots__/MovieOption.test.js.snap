// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieOption /> component tests should render with given state from Redux store 1`] = `
<div>
  <div
    className="sc-gzVnrw czncSY"
  >
    <div>
      <h1
        className="sc-dnqmqq fPpAMb"
      />
      <img
        alt="movieImage"
        className="sc-htoDjs eQCEHt"
      />
    </div>
    <div>
      <button
        className="sc-bZQynM fDFfcI"
        onClick={[Function]}
      >
        🤲 Skip
      </button>
      <button
        className="sc-bZQynM fDFfcI"
        onClick={[Function]}
      >
        👍 Like
      </button>
    </div>
  </div>
</div>
`;
